### ISSUE TITLE
Fix from sinonjs/sinon#650 was never tested

### ISSUE THREAD

**Issue by peterjwest** [sinonjs/sinon-test#3]
The issue sinonjs/sinon/issues/650 was fixed, but never had a test against it. I suggested this test could be used:

```
describe('sinon.test', function() {
  it('should forward exceptions from the test method', function(done) {
    try {
      sinon.test(function(done) {
        expect(false).to.be.true;
        done();
      })();
    }
    catch(e) {
      expect(e).to.equal(
        'AssertionError: expected false to be true'
      );
    }
  });
});
```

**Comment by fatso83** [comment]
Actually sinonjs/sinon#650 was never fixed as sinonjs/sinon#680 was never merged (due to no test). The changes in sinonjs/sinon#680 need to be resubmitted as a PR against this repo, integrating your fix as well.

**Comment by fatso83** [comment]
@peterjwest : could you spare a few minutes to resubmit Morgan's patch in sinonjs/sinon#680 to this repo? We are kind of swamped atm. Then you could just add your test as an extra commit on top of that. 

Would be most appreciated :bowtie:

**Comment by peterjwest** [comment]
Ok, can do.

**Comment by RobertWHurst** [comment]
Fixed by #11.
